### ISSUE TITLE
New Function question

### ISSUE THREAD

**Issue by bahynt** [garris/BackstopJS#1015]
Can we add a button in the report in order to approve the curent test screen to be the reference image for next test?

**Comment by garris** [comment]
That would be fantastic.  What's the plan?

**Comment by bahynt** [comment]
Is this feature possible? 

I mean something like this: In the report, there will be a button says "Approve". 
when we click it, it will send an ajax request to a API that we will create. And this API accept the parameters and execute the approve command with all the parameters. 

Then the next scan will be the latest updated version. 

What do you think? Is it possible? I am not so good at programing, I would love to have your idea about this.

**Comment by garris** [comment]
Yes. It is totally possible.  It is something I've wanted since version 2 but always had bigger priorities.

It would probably require...
- enhance existing `backstop remote` express app with new route to handle running `backstop approve` under correct config and $cwd
- create api call in UI and trigger with button

Do you know how to work with react and express?

**Comment by bahynt** [comment]
I am new to nodejs, but I can try

**Comment by garris** [comment]
If you are going to attempt this, please document your approach here first.   Call out your entry points and how your modify or add to any flows.   This could be a great way for you to learn node if you're interested in getting into it.  But it's also a pretty complex app and the `backstop remote` flow is currently undocumented.   So ok to back off if it's overwhelming.  Cheers.
